src/main/vcp_hal/usbd_cdc_interface.c

```diff
@@ -201,7 +215,12 @@ static int8_t CDC_Itf_Control (uint8_t cmd, uint8_t* pbuf, uint16_t length)
     break;
 
   case CDC_SET_CONTROL_LINE_STATE:
-    /* Add your code here */
+    // If a callback is provided, tell the upper driver of changes in DTR/RTS state
+    if (pbuf && (length == sizeof (uint16_t))) {
```

Comment:
It would be cleaner to build aligned linestate here. Also this code ignores USB endianness

Reply:
You mean to interpret DTR/RTS encoding here? Endianness isn't about to change. We could add a load of endian aware macros around all code processing data to/from the wire, but we're not about to change endianness, so is this really worth the effort?

Reply:
No, just implement is the same as `LineCoding`:
```
LineState = (uint16_t)(pbuf[0] | (pbuf[1] << 8);
```

Reply:
A lot of IO code in BF is endianness aware ... In this case it's mainly to match `LineCoding`